src/topics/troubleshooting.md

```diff
@@ -0,0 +1,70 @@
+# Troubleshooting
+
+In this section you will find ways to troubleshoot when you have problems executing CWL.
+We focus on `cwltool` here but some of these techniques may apply to other CWL Runners.
+
+## Run `cwltool` with `cachedir`
+
+You can use the `--cachedir` option when running a workflow to tell `cwltool` to
+cache intermediate files (files that are not input nor output files, but created
+while your workflow is running). By default, these files are created in a
+temporary directory but writing them to a separate directory makes accessing
+them easier.
+
+In the following example `troubleshooting-wf1.cwl` we have two steps, `step_a` and `step_b`.
+These two steps are executed in order (we enforce it, see note below). The first step,
+`step_a`, executes the `touch` command. The second step, `step_b`, **has a typo**,
+where instead of also executing the `touch` command it tries to execute `ouch`, which
+fails.
+
+```{literalinclude} /_includes/cwl/troubleshooting/troubleshooting-wf1.cwl
+:language: cwl
+:name: "`troubleshooting-wf1.cwl`"
+:caption: "`troubleshooting-wf1.cwl`"
+:emphasize-lines: 25
+```
+
+```{note}
+The CWL Standard does not guarantee the execution order of Workflow Steps. They can
+be executed in any arbitrary order, or in paralell. So the in the `troubleshooting-wf1.cwl`
+CWL document we enforce the order by chaining the output of `step_a` into an input
+of `step_b`.
+```
```

Comment:
Can we use an example workflow that doesn't need this artificial ordering of the steps?

Reply:
Done. I tried to use `echo "Hello World" | rev", introducing the typo in "rev" as "revv". I think it is simpler than the previous example with artificial ordering, but let me know if it would be good to further simplify it or modify something else :+1: 

Thanks @mr-c!